test/Microsoft.Azure.WebJobs.Host.EndToEndTests/Scale/ScaleHostEndToEndTests.cs

```diff
@@ -0,0 +1,357 @@
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Host.EndToEndTests.Scale
+{
+    public class ScaleHostEndToEndTests
+    {
+        private const string Function1Name = "Function1";
+        private const string Function2Name = "Function2";
+        private const string Function3Name = "Function3";
+
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public async Task ScaleManager_GetScaleStatusAsync_ReturnsExpected(bool tbsEnabled)
+        {
+            JArray functionMetadata = new JArray
+            {
+                new JObject
+                {
+                    { "type", "testExtensionATrigger" },
+                    { "functionName", Function1Name }
+                },
+                new JObject
+                {
+                    { "type", "testExtensionATrigger" },
+                    { "functionName", Function2Name }
+                },
+                new JObject
+                {
+                    { "type", "testExtensionBTrigger" },
+                    { "functionName", Function3Name }
+                }
+            };
+
+            string hostJson =
+            @"{
+              ""extensions"": {
+                ""testExtensionA"" : {
+                  ""foo"": 16
+                },
+                ""testExtensionB"" : {
+                  ""bar"": 8
+                }
+              }
+            }";
+
+            string hostId = "test-host";
+            var loggerProvider = new TestLoggerProvider();
+
+            IHostBuilder hostBuilder = new HostBuilder();
+            hostBuilder.ConfigureLogging(configure =>
+            {
+                configure.SetMinimumLevel(LogLevel.Debug);
+                configure.AddProvider(loggerProvider);
+            });
+            hostBuilder.ConfigureAppConfiguration((hostBuilderContext, config) =>
+            {
+                // Adding host.json here
+                config.AddJsonStream(new MemoryStream(Encoding.ASCII.GetBytes(hostJson)));
+
+                // Adding app settings
+                config.AddInMemoryCollection(new List<KeyValuePair<string, string>>()
+                {
+                    new KeyValuePair<string, string>("app_setting1", "value1"),
+                    new KeyValuePair<string, string>("app_setting2", "value2")
+                });
+            })
+            .ConfigureWebJobsScale<TestConcurrencyStatusRepository, TestScaleMetricsRepository>(options =>
+            {
+                // configure scale options
+                options.IsTargetScalingEnabled = tbsEnabled;
+                options.CheckTargetScalerEnabled = ts =>
+                {
+                    return true;
+                };
+                options.MetricsPurgeEnabled = false;
+                options.FunctionMetadata = functionMetadata;
```

Comment:
We also want to pass management identities from SC to extension to initiate azure services clients for the scalers.

We can pass override implementation of [AzureComponentFactory](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/extensions/Microsoft.Extensions.Azure/src/AzureComponentFactory.cs) - CreateTokenCredential method (internal implementation is [AzureComponentFactoryImpl](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/extensions/Microsoft.Extensions.Azure/src/Internal/AzureComponentFactoryImpl.cs)) and use it to create the clients.

To pass various parameters we can add `IDictionary<object, object> Properties`; to the option class, it can be helpfull if extend methods.
Like for [HostBuilderContext](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.hosting.hostbuildercontext.properties?view=dotnet-plat-ext-7.0#microsoft-extensions-hosting-hostbuildercontext-properties)

We need to pass all the identities to all the extensions (like `FunctionMetadata`), that's why I was proposing to call a configure method for each trigger.